src/Clock.js

```diff
@@ -0,0 +1,103 @@
+import React, { Component } from "react";
+
+export default class Clock extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      time: new Date(),
+      hours: true,
+      ampm: false,
+      hour: 12,
+      minute: '00',
+      bn: 'am'
+      };
```

Comment:
The names are confusing. I understand that `hours` is boolean to switch between picking hours and minutes, but I had to read the entire code to understand.
For booleans, it's common to add `is` before the name, such as: `isPickingHours` and `isAM`. 
Also, using abbreviations or short names for variables doesn't bring you far. I don't know what `bn` stands for. Can you make the name more descriptive?

Reply:
isPickingHours: true,
isAm: false,
setHour: 12,
setMinute: '00',
setAmOrPm: 'am'